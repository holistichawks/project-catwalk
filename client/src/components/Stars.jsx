import React from 'react';
import styles from './starsStyle.css';

const Stars = ({ numStars }) => {
  const numFilledStars = Math.floor(numStars);
  const decimal = Number((numStars - numFilledStars).toFixed(2)) * 100;
  const numQuarters = Math.round(decimal / 25);
  let numEmptyStars = 0;
  const array = [];
  for (let i = 0; i < numFilledStars; i++) {
    array.push('stars/filledStar.png');
  }
  if (numQuarters === 0) {
    numEmptyStars = 4 - numFilledStars;
  } else {
    numEmptyStars = 4 - numFilledStars - 1;
    if (numQuarters === 1) {
      array.push('stars/quarterStar.png');
    } else if (numQuarters === 2) {
      array.push('stars/halfStar.png');
    } else if (numQuarters === 3) {
      array.push('stars/threeQuarterStar.png');
    }
  }
  for (let j = 0; j < numEmptyStars; j++) {
    array.push('stars/emptyStar.png');
  }

  return (
    <div>
      {array.map((starFile) => <img className={styles.star} src={`${starFile}`} alt={starFile} />)}
    </div>
  );
};

export default Stars;
