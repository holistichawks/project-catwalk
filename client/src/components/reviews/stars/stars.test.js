import Stars from './stars.jsx';

import { shallow } from 'enzyme';

describe('<Star Component />', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<Stars/>);
    expect(component).toMatchSnapshot();
  });
});