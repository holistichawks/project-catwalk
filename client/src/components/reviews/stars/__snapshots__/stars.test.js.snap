// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Star Component /> should render correctly with no props 1`] = `
<h1>
  Hello I'm a Star
</h1>
`;
